### DISCUSSION TITLE
bevy_asset undeclared imports

### DISCUSSION THREAD

**Discussion by IvoteSligte** [bevyengine/bevy#7135] [Q&A] [answered]
## Bevy version

The release number or commit hash of the version you're using.

## \[Optional\] Relevant system information

Rust version: cargo 1.68.0-nightly (8c460b223 2023-01-04)
Operating system: Linux Mint with kernel 5.15.0-57-generic

## What you did

I tried to use bevy_kira_audio to play an mp3 audio file.
Bevy has mp3 support disabled by default so I enabled the cargo feature.

## What went wrong

As soon as I enabled the mp3 feature I got a series of 'undeclared import' errors originating from the bevy_asset (0.9.1) crate.

## Additional information

- Removing the changes that I made since importing `bevy_kira_audio` and enabling the mp3 feature did not fix the problem.
- I only get these errors in release mode (cargo run --release).
- The bevy_asset crate with version 0.9.0 compiles correctly.

Error log:

error[E0433]: failed to resolve: use of undeclared crate or module `bevy_kira_audio`
 --> /home/{username}/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_asset-0.9.1/src/asset_server.rs:9:5
   |
9 | use bevy_kira_audio::prelude::*;
   |     ^^^^^^^^^^^^^^^ use of undeclared crate or module `bevy_kira_audio`

error[E0432]: unresolved imports `crate::Asset`, `crate::AssetIo`, `crate::AssetIoError`, `crate::AssetLifecycle`, `crate::AssetLifecycleChannel`, `crate::AssetLifecycleEvent`, `crate::AssetLoader`, `crate::Assets`, `crate::Handle`, `crate::HandleId`, `crate::HandleUntyped`, `crate::LabelId`, `crate::LoadContext`, `crate::LoadState`, `crate::RefChange`, `crate::RefChangeChannel`, `crate::SourceInfo`, `crate::SourceMeta`
 --> /home/{username}/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_asset-0.9.1/src/asset_server.rs:3:5
   |
3 |     Asset, AssetIo, AssetIoError, AssetLifecycle, AssetLifecycleChannel, AssetLifecycleEvent,
   |     ^^^^^  ^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^
4 |     AssetLoader, Assets, Handle, HandleId, HandleUntyped, LabelId, LoadContext, LoadState,
   |     ^^^^^^^^^^^  ^^^^^^  ^^^^^^  ^^^^^^^^  ^^^^^^^^^^^^^  ^^^^^^^  ^^^^^^^^^^^  ^^^^^^^^^
5 |     RefChange, RefChangeChannel, SourceInfo, SourceMeta,
   |     ^^^^^^^^^  ^^^^^^^^^^^^^^^^  ^^^^^^^^^^  ^^

**Comment by DJMcNab** [comment]
The error messages you have strangely suggest that the crates registry has been edited? That error message shows a different line 9 than the file actually has (looking at https://docs.rs/crate/bevy_asset/latest/source/src/asset_server.rs)

We cannot support editing the crates within the crates registry cache of cargo, as that is not supported by cargo.

But perhaps running `cargo clean` might fix your issues.

**Reply by IvoteSligte** [reply]
I apologise, I edited the cargo.toml file of bevy_asset in a pathetic attempt to resolve the problem, which seemed to do nothing so I forgot about it. (I added the bevy_kira_audio crate in response to the first error message.) I will try `cargo clean` as soon as I have the time, thank you for the help!

**Comment by nicopap** [comment, marked as answer]
You shouldn't enable the bevy `mp3` feature if you are using `bevy_kira_audio`, instead you should enable `bevy_kira_audio`'s `mp3` feature. Regardless, @DJMcNab answer should fix your current issue.

**Reply by IvoteSligte** [reply]
That makes a lot of sense, thank you!
